administrator/components/com_guidedtours/src/Model/TourModel.php

```diff
@@ -0,0 +1,340 @@
+
+            $table = $this->getTable('Step');
+
+            $table->id = 0;
+            $table->title = 'COM_GUIDEDTOURS_BASIC_STEP';
+            $table->description = '';
+            $table->tour_id = $tour_id;
+
+            // ---changes
+            $table->published = 1;
+
+            $table->store();
+        }
+
+        return $result;
+    }
+
+    /**
+     * Abstract method for getting the form from the model.
+     *
+     * @param   array    $data      Data for the form.
+     * @param   boolean  $loadData  True if the form is to load its own data (default case), false if not.
+     *
+     * @return \JForm|boolean  A JForm object on success, false on failure
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function getForm($data = [], $loadData = true)
+    {
+        // Get the form.
+        $form = $this->loadForm(
+            'com_guidedtours.tour',
+            'tour',
+            array(
+                'control'   => 'jform',
+                'load_data' => $loadData
+            )
+        );
+
+        if (empty($form)) {
+            return false;
+        }
+
+        $id = $data['id'] ?? $form->getValue('id');
+
+        $item = $this->getItem($id);
+
+        $canEditState = $this->canEditState((object) $item);
+
+        // Modify the form based on access controls.
+        if (!$canEditState || !empty($item->default)) {
+            if (!$canEditState) {
+                $form->setFieldAttribute('published', 'disabled', 'true');
+                $form->setFieldAttribute('published', 'required', 'false');
+                $form->setFieldAttribute('published', 'filter', 'unset');
+            }
+        }
+
+        $form->setFieldAttribute('created', 'default', Factory::getDate()->toSql());
+        $form->setFieldAttribute('modified', 'default', Factory::getDate()->toSql());
+
+        return $form;
+    }
+
+    /**
+     * Method to get the data that should be injected in the form.
+     *
+     * @return mixed  The data for the form.
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    protected function loadFormData()
+    {
+        // Check the session for previously entered form data.
+        $data = Factory::getApplication()->getUserState(
+            'com_guidedtours.edit.tour.data',
+            []
+        );
+
+        if (empty($data)) {
+            $data = $this->getItem();
+        }
+
+        return $data;
+    }
+    /**
+     * Method to change the default state of one item.
+     *
+     * @param   array    $pk     A list of the primary keys to change.
+     * @param   integer  $value  The value of the home state.
+     *
+     * @return  boolean  True on success.
+     *
+     * @since  __DEPLOY_VERSION__
+     */
+    public function setDefault($pk, $value = 1)
+    {
+        $table = $this->getTable();
+
+        if ($table->load($pk)) {
+            if ($table->published !== 1) {
+                $this->setError(Text::_('COM_WORKFLOW_ITEM_MUST_PUBLISHED'));
```

Comment:
wrong error message - this component doesnt use workflows

Reply:
Removed the function